### PR TITLE
Add script to check required Terraform version

### PR DESCRIPTION
**Pull request by wwwil**
Add script to check required Terraform version and soft link appropriate version.

This [PR to testing](https://github.com/jetstack/testing/pull/259) will mean that multiple Terraform version are available in the test image as `terraform-0.11` and `terraform-0.12`. This script can then link the appropriate one to `terraform`.

If no known version can be established it will exit with `1` to halt the testing.

### CODE REVIEW

File: hack/check-terraform-version.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Make the zonal cluster resource definiton from the regional cluster resource
+# definition. This helps to keep the two definitions the same, except for the
+# presence of the region or zone property.
```
Comment (munnerz) [inline hack/check-terraform-version.sh]: Think this needs updating?

File: hack/verify.sh [hunk 2]
```diff
@@ -23,4 +23,5 @@ set -o nounset
 set -o pipefail
 
 REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+$REPO_ROOT/hack/check-terraform-version.sh
```
Comment (munnerz) [inline hack/verify.sh]: Would it make more sense to make this `verify-terraform-version`?

**Review by simplyzee — APPROVED** [review]

**Pull request merged**